crates/store/re_video/src/decode/ffmpeg.rs

```diff
@@ -151,6 +154,61 @@ impl FfmpegProcessAndListener {
+            });
+        if let Some(sps_unit) = sps_units.next() {
+            if sps_units.next().is_some() {
+                // This is rather strange. Must mean that some pictures refer to one SPS and some to another!
+                // We don't know what to do with this.
+                re_log::trace_once!("Found more than one sequence parameter set (SPS) in the AVCC box of {debug_name}.");
+            } else if let Ok(sps) = H264Sps::try_parse(&sps_unit.bytes[1..]) {
+                re_log::trace!("Successfully parsed SPS for {debug_name}:\n{sps:?}");
+
+                if let Some(mut layout) = sps.pixel_layout() {
+                    // TODO(nathanbabcock/ffmpeg-sidecar#61): ffmpeg-sidecar can't handle this yet. Quite unfortunate since this is the most common case by far!
+                    if layout == crate::decode::YuvPixelLayout::Y_U_V420 {
+                        layout = crate::decode::YuvPixelLayout::Y_U_V422;
+                    }
+                    pixel_format = PixelFormat::Yuv {
+                        layout,
+                        // Unfortunately the color range is an entirely different thing to parse as it's part of optional Video Usability Information (VUI).
+                        //
+                        // We instead just always tell ffmpeg to give us full range, see`-color_range` below.
+                        // Note that yuvj4xy family of formats fulfill the same function. They according to this post
+                        // https://www.facebook.com/permalink.php?story_fbid=2413101932257643&id=100006735798590
+                        // they are still not quite passed through everywhere. So we'll just use both.
+                        range: crate::decode::YuvRange::Full,
+                        // Again, instead of parsing this out we tell ffmpeg to give us BT.709.
+                        coefficients: crate::decode::YuvMatrixCoefficients::Bt709,
+                    };
+                    ffmpeg_pix_fmt = match layout {
+                        crate::decode::YuvPixelLayout::Y_U_V444 => "yuvj444p",
+                        crate::decode::YuvPixelLayout::Y_U_V422 => "yuvj422p",
+                        crate::decode::YuvPixelLayout::Y_U_V420 => "yuvj420p",
+                        crate::decode::YuvPixelLayout::Y400 => "gray",
+                    };
+                } else {
+                    re_log::warn_once!(
+                        "Failed to parse pixel layout from sequence parameter set (SPS) for {debug_name}. `chroma_format_idc` was {}.",
+                        sps.chroma_format_idc
+                    );
+                }
+            } else {
+                re_log::warn_once!(
+                    "Failed to parse the sequence parameter set (SPS) for {debug_name}."
+                );
+            }
+        } else {
+            re_log::warn_once!("Expected at least one sequence parameter set (SPS) in the AVCC box of {debug_name}, but found none.");
+        };
+
```

Comment:
It feels like this whole code bloc could be moved to its own function, perhaps one that returns errors that the caller can can `re_log::warn_once`